index.js

```diff
@@ -1,16 +1,95 @@
 const functions = require('@google-cloud/functions-framework');
-const axios = require("axios");
+const webpush = require('web-push');
+const Knex = require('knex');
 
+webpush.setVapidDetails(
+  process.env.VAPID_SUBJECT || 'mailto:milena@near.org',
+  process.env.VAPID_PUBLIC_KEY || "BCZN5uqYMBZ2VCV3y0F0emodyYRGyt5JTgfIIzYVXIHKSBwuG0kb0NpPA-DM4nfmfRFiFu-MpKS2eNG7bhxQWn0",
+  process.env.VAPID_PRIVATE_KEY || "f1qlHM-Wsq9m6Z_WwQPPGL4zHyS2fKCoZFXOu2n5Dao"
+);
+
+
+// For local testing
+// const DB_USER = "postgres";
+// const DB_PASS = "postgres";
+// const DB_NAME = "notifications-db";
+// const INSTANCE_HOST = "localhost";
+// const DB_PORT = 8087;
+
+const createTcpPool = async config => {
+    return Knex({
+      client: 'pg',
+      connection: {
+        user: process.env.DB_USER || DB_USER,
+        password: process.env.DB_PASS || DB_PASS,
+        database: process.env.DB_NAME || DB_NAME,
+        host: process.env.INSTANCE_HOST || INSTANCE_HOST,
+        port: process.env.DB_PORT || DB_PORT,
+      },
+      ...config,
+    });
+};
+
+const insertNotification = async (pool, notification) => {
+    try {
+      return await pool('Notification').insert({
+          id: notification.id,
+          block_height: notification.blockHeight,
+          initiated_by: notification.initiatedBy,
+          item_type: notification.itemType,
+          message: notification.message,
+          path: notification.path,
+          receiver: notification.receiver,
+          value_type: notification.valueType,
+          sent_at: new Date(),
+        });
+    } catch (err) {
+      throw Error(err);
+    }
+  };
+
+const getSubscription = async (pool, accountId) => {
+    try {
+      return await pool('Subscription')
+      .select('push_subscription_object')
+      .where('account', accountId)
```

Comment:
@milenakrawczyk is the expectation that there will only be 1 subscription per user?

Reply:
Yes, I wanted to implement the simplest flow for now just to test and see how it behaves but I may extend it to encompass multiple subscriptions per user. What would be you preference?

Reply:
We should have 1:many account-to-push subscriptions for the MVP